rest-server/src/main/java/org/opennms/horizon/server/service/NodeStatusService.java

```diff
@@ -196,46 +202,48 @@ private NodeReachability transformToNodeReachability(long id, TimeSeriesQueryRes
             var roundedValue = Math.min(reachability, 100.0);
             return new NodeReachability(id, roundedValue);
         }
-        return new NodeReachability(id, 0);
+        return new NodeReachability(id, 0.0);
     }
 
-    public NodeResponseTime getNodeAvgResponseTime(NodeDTO node, Integer timeRange, TimeRangeUnit timeRangeUnit, ResolutionEnvironment env) {
+    public Mono<NodeResponseTime> getNodeAvgResponseTime(NodeDTO node, Integer timeRange, TimeRangeUnit timeRangeUnit, ResolutionEnvironment env) {
         IpInterfaceDTO ipInterface = getPrimaryInterface(node);
         Map<String, String> labels = new HashMap<>();
         labels.put(NODE_ID_KEY, String.valueOf(node.getId()));
         labels.put(MONITOR_KEY, Constants.DEFAULT_MONITOR_TYPE);
         labels.put(INSTANCE_KEY, ipInterface.getIpAddress());
 
-        var future = tsdbMetricsService.getMetric(env,
-            AVG_RESPONSE_TIME, labels, timeRange, timeRangeUnit).toFuture();
         try {
-            var timeSeriesResult = future.join();
-            return transformToNodeResponseTime(node.getId(), timeSeriesResult);
+            return tsdbMetricsService.getMetric(env,
+                AVG_RESPONSE_TIME, labels, timeRange, timeRangeUnit).map(result ->
+                transformToNodeResponseTime(node.getId(), result));
         } catch (Exception e) {
             LOG.warn("Failed to get response time for node id {}", node.getId(), e);
         }
-        return new NodeResponseTime(node.getId(), 0);
+        return Mono.just(new NodeResponseTime(node.getId(), 0.0));
     }
 
-    public TopNNode getTopNNode(NodeDTO nodeDTO, Integer timeRange, TimeRangeUnit timeRangeUnit, ResolutionEnvironment env) {
-         var nodeReachability = getNodeReachability(nodeDTO, timeRange, timeRangeUnit, env);
-         var nodeResponseTime = getNodeAvgResponseTime(nodeDTO, timeRange, timeRangeUnit, env);
-         var topNNode = new TopNNode();
-         topNNode.setNodeLabel(nodeDTO.getNodeLabel());
-         topNNode.setLocation(nodeDTO.getLocation());
-         topNNode.setReachability(nodeReachability.getReachability());
-         topNNode.setAvgResponseTime(nodeResponseTime.getResponseTime());
-         return topNNode;
+    public Mono<TopNNode> getTopNNode(NodeDTO nodeDTO, Integer timeRange, TimeRangeUnit timeRangeUnit, ResolutionEnvironment env) {
+        var nodeReachability = getNodeReachability(nodeDTO, timeRange, timeRangeUnit, env);
+        var nodeResponseTime = getNodeAvgResponseTime(nodeDTO, timeRange, timeRangeUnit, env);
+        var result = nodeReachability.zipWith(nodeResponseTime);
```

Comment:
Because these items return a container type rather than the underlying type, I think it would make sense to use the explicit type here:

```suggestion
        Mono<NodeReachability> nodeReachability = getNodeReachability(nodeDTO, timeRange, timeRangeUnit, env);
        Mono<NodeResponseTime> nodeResponseTime = getNodeAvgResponseTime(nodeDTO, timeRange, timeRangeUnit, env);
        Mono<Tuple2<NodeReachability, NodeResponseTime>> result = nodeReachability.zipWith(nodeResponseTime);
```

Otherwise it's not immediately clear that we're working with `Mono`s, or even the `Mono<Tuple2<>>`